dags/s3_xml_import_pipeline.py

```diff
@@ -136,30 +134,82 @@ def update_prev_run_id_var_val(**context):
 
 
 def etl_new_ejp_xml_files(**context):
-    dep_env = os.getenv(
-        DEPLOYMENT_ENV_ENV_NAME, DEFAULT_DEPLOYMENT_ENV_VALUE
-    )
-    dag_context = context["ti"]
-    data_config_dict = dag_context.xcom_pull(
-        key="data_config",
-        task_ids="Should_Remaining_Tasks_Execute"
-    )
-    data_config = eJPXmlDataConfig(data_config_dict, dep_env)
-
+    data_config = get_config()
     obj_pattern_with_latest_dates = (
-        get_stored_state(
+        get_stored_ejp_xml_processing_state(
             data_config,
             get_default_initial_s3_last_modified_date()
         )
     )
+    matching_file_metadata_iter = etl_s3_object_pattern(
+        data_config,
+        obj_pattern_with_latest_dates,
+        data_config.s3_bucket,
+    )
+
+    for matching_file_metadata, object_key_pattern in matching_file_metadata_iter:
+        object_key = matching_file_metadata.get(
+            named_literals.S3_FILE_METADATA_NAME_KEY
+        )
+        etl_ejp_xml_zip(
+            data_config, object_key,
+        )
+
+        updated_obj_pattern_with_latest_dates = (
+            update_object_latest_dates(
+                obj_pattern_with_latest_dates,
+                object_key_pattern,
+                matching_file_metadata.get(
+                    named_literals.S3_FILE_METADATA_LAST_MODIFIED_KEY
+                )
+            )
+        )
+        update_state(
+            updated_obj_pattern_with_latest_dates,
+            data_config.state_file_bucket,
+            data_config.state_file_object
+        )
+
+
+def load_temp_ejp_json_files_to_bq(**context):
+    data_config = get_config()
+    batch_size_limit = 5000
```

Comment:
`5000` seems to be slightly low. But maybe there is a reason?

Reply:
no particular reason, 20000?

Reply:
The previous XML pipeline was uploading everything. I believe the file size limit is [5 GB uncompressed](https://cloud.google.com/bigquery/quotas).

I guess the main benefit of using a lower batch size would be to have partial uploads if something is failing. But not sure that is something we expect. Maybe use 100k?